README.md

```diff
@@ -0,0 +1,89 @@
+# MoJ S3 Virus Scanner
+
+A service to scan viruses automatically in S3 buckets.
+
+This project is a service to provide automatic virus scanning for files in AWS
+S3 buckets. The intended way to use this service is via an event attached to an
+S3 bucket which calls a Lambda function, which in turn makes an HTTP request to
+an instance of this application which then performs the scan and updates tags on
+the file in S3 to indicate whether it has passed or failed the virus scan.
+
+![Architecture Diagram](docs/architecture.png)
+
+When a file is uploaded to an S3 bucket, a on-create event is triggered and runs
+a lambda function which begins the virus scan on the newly created file. Once
+the file is scanned it is tagged according to whether it passed the scan or not,
+and these tags can be used to determine whether the file is safe to present to
+the user or not.
+
+This approach is an adaptation of the [bucket-antivirus-function approach
+presented by Upside Travel]
+(https://github.com/upsidetravel/bucket-antivirus-function), but allows the
+scanning to be performed much more quickly (1s-2s vs 10s-20s). It is
+particularly well suited to use with services that do direct uploads to S3, e.g.
+using Dropzone, but works equally well with any service that needs virus scanning.
+
+## Scanning Application
+
+The scanning is preformed by the application in this repository which provides
+the endpoint `/scan` to perform the scan. It takes the request params `bucket`
+and `key` to specify which object to scan and requires permissions to access the
+bucket provided.
+
+## Lambda Function
+
+To perform the scan a lambda function must be created as a bit of glue between
+the S3 bucket and the scanning application. The steps to create this are:
+
+### Creation
+
+1. Go to the Lambda section in the Amazon console.
+2. Create a new function:
+   * Select "Author from scratch"
+   * Give the function name e.g. moj-s3-virus-scan
+   * Select a Python 3.x runtime
+   * Select "Create new role from template"
+   * Give the role a name, e.g. moj-s3-virus-scan-role
```

Comment:
Don't find this example clear

Reply:
In discussing IRL with @beccapearce, we've said that this is not clear what this should be, esp since there's probably already a role at the MoJ called this, etc. The documentation should clarify when you'd want to create a new lambda function / role, i.e. when setting this up in your organisation. That will be made more clear once we figure out the model for how to integrate this service with your application and document that.